pkgs/development/lisp-modules/import/main.lisp

```diff
@@ -0,0 +1,42 @@
+(defpackage org.lispbuilds.nix/main
+  (:use :common-lisp
+        :org.lispbuilds.nix/database/sqlite
+        :org.lispbuilds.nix/repository/quicklisp
+        :org.lispbuilds.nix/api))
+
+(in-package org.lispbuilds.nix/main)
+
+(defun resource (name type)
+  (make-pathname
+   :defaults (asdf:system-source-directory :org.lispbuilds.nix)
+   :name name
+   :type type))
+
+(defvar *sqlite*
+  (make-instance
+   'sqlite-database
+   :init-file (resource "init" "sql")
+   :url "packages.sqlite"))
+
+;; Check http://beta.quicklisp.org/dist/quicklisp.txt for updates
+(defvar *quicklisp*
+  (make-instance
+   'quicklisp-repository
+   :dist-url
+   "https://beta.quicklisp.org/dist/quicklisp/2023-02-15/"))
+
+(defun run-importers ()
+  (ignore-errors (delete-file "packages.sqlite"))
+  (import-lisp-packages *quicklisp* *sqlite*)
+  (format t "Imported packages from quicklisp to ~A~%"
+          (truename "packages.sqlite")))
+
+(defun gen-nix-file ()
+  (database->nix-expression *sqlite* "imported.nix")
+  (format t "Dumped nix file to ~a~%"
+          (truename "imported.nix")))
+
+(defun main ()
```

Comment:
Maybe add support for using the latest quicklisp release if a command-line flag is passed? 

If we already want `@var@` substitutions, we could at least control the version used via something Nix-visible. Or just keep the current quicklisp package, and use version from there.

Reply:
> Maybe add support for using the latest quicklisp release if a command-line flag is passed?

This sounds better to me, But also in my opinion updating the file once in a while is not bad. But it's not too hard to implement, so I won't oppose.

Do you think it should be added now or in a future patch?

Reply:
I think having a version pin that is not complete enough for fixed-output derivation, not auto-updatable, not command-line controllable, and not visible from Nix, oh and also it is in the middle of a source file with some meaningful code too — it is too much. If we update is in a source file, while not just keep the quicklisp package and take files from there? And then an auto-updater can be added some day.

Reply:
OK, I made it so it always takes the latest Quicklisp version https://github.com/NixOS/nixpkgs/pull/218870/commits/57df1050133e4591ba2b0544e15f7a54462ac405

Reply:
Do you think that solves the problem?

Reply:
I think for practical purposes yes, there are some people with huge global ideas who will be unhappy about lack of the pinned options, but I am currently busy conspiciously not buying their bridge anyway.